airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/abstract_stream.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, Mapping, Optional
+
+from airbyte_cdk.models import AirbyteStream
+from airbyte_cdk.sources.streams.concurrent.availability_strategy import StreamAvailability
+from airbyte_cdk.sources.streams.concurrent.partitions.record import Record
+from deprecated.classic import deprecated
+
+
+@deprecated("This class is experimental. Use at your own risk.")
+class AbstractStream(ABC):
+    """
+    AbstractStream is an experimental interface for streams developed as part of the Concurrent CDK.
+    This interface is not yet stable and may change in the future. Use at your own risk.
+
+    Why create a new interface instead of adding concurrency capabilities the existing Stream?
+    We learnt a lot since the initial design of the Stream interface, and we wanted to take the opportunity to improve.
```

Comment:
Do we have a goal of migrating existing connectors to use the new interface and/or prevent development on the old one?

Reply:
It would be great because I think it would with maintainability, but I don't think we should actively invest into the migration. It can happen as we decide to add concurrency to the source.